### PR TITLE
fix #67 : add replace_axes method to LArray class.

### LINKED ISSUE #67: implement replace_axis
to complement with_axes, we need a way to replace only one axis (or a few axes). The goal is to have something nicer than:
```python
a.with_axes(a.axes.replace(x.products, industries))
```
e.g:
```python
a.replace_axis(x.products, industries)
# or
a.with_axes(products=industries)
```

### PR DESCRIPTION
**Pull request by alixdamman**
with signature similar to rename method (to stay consistent).

### CODE REVIEW

**Comment by alixdamman** [comment]
OK, the case where we pass a list wasn't tested in docttest of rename. 
The error here is with Python 2.7 : 
'list' object has no attribute 'copy'
I need to change implementation of rename and replace_axes

**Comment by gdementen** [comment]
I **wonder** (not sure at all) whether we should add the case where a single list of Axis is passed to replace_axes/rename for the case where we want to replace all axes, in which case, we could deprecate with_axes. The thing is that when having replaces_axes, it **seems** like with_axes is one too many method. What do you think?

**Comment by alixdamman** [comment]
I would go for setting with_axes as depecrated.

**Comment by gdementen** [comment]
We still need a credible upgrade path. Previously with_axes was mostly used to align axes with those of another array: `arr1.with_axes(arr2.axes)`. The new syntax seems to be annoying when you want to replaces all axes, or do you see something better than:
```python
arr1.replace_axes(zip(arr1.axes, arr2.axes))
```
Hence I wanted to allow axes_to_replace to be a list of Axis in addition to a list of tuples. In that case the list must be exactly the same length as self.axes. As I said I am unsure it is a good idea, but deprecating with_axes without providing a nice syntax for that use case seems like a bad move.

Maybe allowing new_axes to be a list (by the name of the argument one might think it is allowed anyway), and using: 
```python
arr1.replace_axes(arr1.axes, arr2.axes)
```
would be good enough (honestly, I do not really like it)?

**Comment by alixdamman** [comment]
I chose to have a function signature as close as possible to one of rename method but you get a point here. 
I agree to allow axes_to_replace to be a list of Axis and to check its length in that case.

File: larray/core.py [hunk 1]
```diff
@@ -3884,22 +3884,34 @@ def nonzero(self):
...
-        (int, str or  Axis), optional
-            New axes.
+        axes_to_replace : axis ref or dict {axis ref: axis} or
+                  list of tuple (axis ref, axis) or list of Axis
+            Axes to replace. If a single axis reference is given,
+            the `new_axes` argument must be used. If a list of
+            Axis is given, all axes will be replaced by the
+            new ones. In that case, the number of new axes must
+            match the number of the old ones.
+        new_axes : Axis
```
Comment (gdementen) [inline larray/core.py]: we should either rename this to new_axis, or allow a list of Axis

Comment (alixdamman) [inline larray/core.py]: 'new_axis' it will be.

**Comment by gdementen** [comment]
LGTM, merging...

**Pull request merged**